src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -36,6 +36,18 @@ public boolean contains(Person toCheck) {
         return internalList.stream().anyMatch(toCheck::isSamePerson);
     }
 
+    public Person getPerson(String nricString) {
+        requireAllNonNull(nricString);
+        Nric nric = new Nric(nricString);
+        Person person = new Person(nric);
+        for (int i=0; i < internalList.size(); i++) {
+            if(internalList.get(i).isSamePerson(person)) {
+                return internalList.get(i);
+            }
+        }
+        return null;
+    }
+
```

Comment:
There is a uniqueNricMap that allows you to retrieve the person from nricString in O(1) haha, but this is fine too